### PR TITLE
Update discogapic 

### PR DESCRIPTION
**Pull request by andreamlin**
Discogapic pipeline needs to be updated to reflect gax-java changes.
Mostly just fixing imports.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java [hunk 1]
```diff
@@ -676,6 +675,9 @@ private void addSettingsImports(InterfaceContext context) {
     typeTable.saveNicknameFor("javax.annotation.Generated");
     typeTable.saveNicknameFor("org.threeten.bp.Duration");
 
+    // TODO(andrealin): remove this from httpjson
+    typeTable.saveNicknameFor("com.google.api.gax.grpc.GrpcExtraHeaderData");
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java]: I don't quite understand this TODO - this class isn't even accessible from httpjson.

Comment (andreamlin) [reply src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java]: Ok, this import has been applied only to proto-input surfaces.

**Comment by andreamlin** [comment]
PTAL

File: src/main/resources/com/google/api/codegen/java/settings.snip [hunk 2]
```diff
@@ -162,8 +162,11 @@
   public static ApiClientHeaderProvider.Builder defaultApiClientHeaderProviderBuilder() {
     return ApiClientHeaderProvider.newBuilder()
         .setGeneratorHeader(DEFAULT_GAPIC_NAME, getGapicVersion())
-        .setApiClientHeaderLineKey("x-goog-api-client")
-        .addApiClientHeaderLineData(GrpcExtraHeaderData.getXGoogApiClientData());
+        @if xsettingsClass.transportProtocol == "GRPC"
+          .setApiClientHeaderLineKey("x-goog-api-client")
```
Comment (garrettjonesgoogle) [inline src/main/resources/com/google/api/codegen/java/settings.snip]: We want the header line key for httpjson also.

Comment (andreamlin) [reply src/main/resources/com/google/api/codegen/java/settings.snip]: Can this be hardcoded as HTTP/1.1, given how stable that config will be?

Comment (garrettjonesgoogle) [reply src/main/resources/com/google/api/codegen/java/settings.snip]: I just meant that x-goog-api-client should be included for discogapic clients. I don't fully understand your question.

Comment (andreamlin) [reply src/main/resources/com/google/api/codegen/java/settings.snip]: Oh gotcha. Moved the x-goog-api-client line out of the @if block.

**Comment by andreamlin** [comment]
PTAL

**Pull request merged**